src/transformStateWithClones.js

```diff
@@ -8,6 +8,28 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateCopy[keyToRemove];
+        }
+        break;
+      case 'clear':
+        stateCopy = {};
```

Comment:
don't create new empty object, it's better to clear existing object instead of creating a new one